src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -240,25 +374,92 @@ inline DocumentStatisticsAnnotationData::DocumentStatisticsAnnotationData(Freque
 // |
-// |  Details::DocumentStatisticsTrainingOnlyPolicy
+// |  DocumentStatisticsEstimator
 // |
 // ----------------------------------------------------------------------
-inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(void) :
-    _totalNumDocuments(0) {
+//Would it be better to place default values during pipeline chaining(countvectorizer and tfidfvectorizer)? if so this ctor will show no default values
+template <size_t MaxNumTrainingItemsV>
+DocumentStatisticsEstimator<MaxNumTrainingItemsV>::DocumentStatisticsEstimator(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex,
+                                                                               StringDecorator decorator, nonstd::optional<IndexMap> vocabulary, 
+                                                                               nonstd::optional<std::uint32_t> maxFeatures, std::float_t minDf, std::float_t maxDf) :
+    BaseType(
+        std::move(pAllColumnAnnotations),
+        std::move(colIndex),
+        true,
+        std::move(decorator),
+        std::move(vocabulary),
+        std::move(maxFeatures),
+        std::move(minDf),
+        std::move(maxDf)
+    ) {
 }
 
-inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(StringDecorator decorator) :
+// ----------------------------------------------------------------------
+// |
+// |  Details::DocumentStatisticsTrainingOnlyPolicy
+// |
+// ----------------------------------------------------------------------
+inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(StringDecorator decorator, nonstd::optional<IndexMap> vocabulary, 
+                                                                                           nonstd::optional<std::uint32_t> maxFeatures, std::float_t minDf, 
+                                                                                           std::float_t maxDf) :
     _stringDecoratorFunc(
         [&decorator](void) -> StringDecorator & {
-            if(!decorator)
-                throw std::invalid_argument("decorator");
-
+            //temporary mute this validate
+            // if(!decorator)
+            //     throw std::invalid_argument("decorator");
             return decorator;
         }()
     ),
+    _vocabulary(
+        std::move(
+            [&vocabulary](void) -> nonstd::optional<IndexMap> & {
+                if(vocabulary.has_value() && vocabulary->empty()) 
+                    throw std::invalid_argument("vocabulary");
+
+                return vocabulary;
+            }()
+        )
+    ),
+    _maxFeatures(
+        std::move(
+            [&maxFeatures](void) -> nonstd::optional<std::uint32_t> & {
+                if(maxFeatures.has_value() && *maxFeatures == 0)
+                    throw std::invalid_argument("maxFeaturesLowerBound");
+
+                return maxFeatures;
+            }()
+        )
+    ),
+    _minDf(
```

Comment:
Are NaN values valid for _minDf and _maxDf?

Reply:
We only allow these two parameters in the range of [0.0f, 1.0f]